var/spack/repos/builtin/packages/julia/package.py

```diff
@@ -3,206 +3,184 @@
-    # depends_on('gmp')
-    # depends_on('libgit')
-    # depends_on('mpfr')
-    # depends_on('openblas')
-    # depends_on('pcre2')
-
-    # ARPACK: Requires BLAS and LAPACK; needs to use the same version
-    # as Julia.
-
-    # BLAS and LAPACK: Julia prefers 64-bit versions on 64-bit
-    # systems. OpenBLAS has an option for this; make it available as
-    # variant.
-
-    # FFTW: Something doesn't work when using a pre-installed FFTW
-    # library; need to investigate.
-
-    # GMP, MPFR: Something doesn't work when using a pre-installed
-    # FFTW library; need to investigate.
-
-    # LLVM: Julia works only with specific versions, and might require
-    # patches. Thus we let Julia install its own LLVM.
-
-    # Other possible dependencies:
-    # USE_SYSTEM_OPENLIBM=0
-    # USE_SYSTEM_OPENSPECFUN=0
-    # USE_SYSTEM_DSFMT=0
-    # USE_SYSTEM_SUITESPARSE=0
-    # USE_SYSTEM_UTF8PROC=0
-    # USE_SYSTEM_LIBGIT2=0
-
-    conflicts('+cxx', when='@:0.6', msg='Variant cxx requires Julia >= 1.0.0')
-
-    conflicts('@:0.7.0', when='target=aarch64:')
-
-    # GCC conflicts
-    conflicts('@:0.5.1', when='%gcc@8:', msg='Julia <= 0.5.1 needs GCC <= 7')
-
-    # Building recent versions of Julia with Intel is untested and unsupported
-    # by the Julia project, https://github.com/JuliaLang/julia/issues/23407.
-    conflicts('@0.6:', when='%intel',
-              msg='Only Julia <= 0.5.x can be built with the Intel compiler.')
-    conflicts('%intel', when='~mkl',
-              msg='Building with the Intel compiler requires the mkl variant '
-              '(+mkl)')
+    version('1.7.0-rc3', sha256='87604d53bc0832045522bc92ee98923fca370ead43f567c0676c6b9ecf1c5134')
+
+    # We've deprecated these versions, so that we can remove them in Spack 0.18
+    # They are still available in Spack 0.17. Julia 0.17.0 is the first version that
+    # can be built enitrely from Spack packages, without a network connection during
+    # the build.
+    for v in [
+        '1.6.3', '1.6.2', '1.6.1', '1.6.0', '1.5.4', '1.5.3', '1.5.2', '1.5.1', '1.5.0',
+        '1.4.2', '1.4.1', '1.4.0', '1.3.1', '1.2.0', '1.1.1', '1.0.0', '0.6.2', '0.5.2',
+        '0.5.1', '0.5.0', '0.4.7', '0.4.6', '0.4.5', '0.4.3'
+    ]:
+        version(v, deprecated=True)
+
+    variant('precompile', default=True, description='Improve julia startup time')
+    variant('openlibm', default=True, description='Use openlibm instead of libm')
+
+    # Note, we just use link_llvm_dylib so that we not only get a libLLVM,
+    # but also so that llvm-config --libfiles gives only the dylib. Without
+    # it it also gives static libraries, and breaks Julia's build.
+    depends_on('llvm version_suffix=jl +link_llvm_dylib ~internal_unwind')
+    depends_on('llvm@12.0.1')  # 13.0.0 not yet supported
+    depends_on('libuv@1.42.0')
+
+    # patches for dependencies
+    depends_on('llvm', patches='llvm7-symver-jlprefix.patch')
+    depends_on('llvm', when='^llvm@12.0.1', patches=patch(
```

Comment:
So technically this is Julia 1.7 and LLVM 12.
1.8 might/will have a different patch file.

But might be ok to ignore for now.

Reply:
Right, as it is I'm basing patches only on the LLVM version, but maybe it should depend on both julia and LLVM version.